src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head = new Node<>(null, null, null);
+    private Node<T> tail = head;
+    private Node<T> previous = tail;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(previous, value, head);
+        previous.next = newNode;
+        head.prev = newNode;
+        previous = newNode;
+        head = newNode.next;
+        if (size == 0) {
+            tail = previous;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, tail);
+            newNode.next = tail;
+            tail.prev = newNode;
+            tail = newNode;
+            size++;
+        } else {
+            checkIndex(index);
+            Node<T> myNode = tail;
+            for (int i = 0; i < index; i++) {
+                if (myNode.next != null) {
+                    myNode = myNode.next;
+                }
+            }
+            Node<T> prevNode = myNode.prev;
+            Node<T> nextNode = myNode;
+            Node<T> newNode = new Node<>(prevNode, value, nextNode);
+            nextNode.prev = newNode;
+            prevNode.next = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> newNode = tail;
+        for (int i = 0; i < index; i++) {
```

Comment:
please move logic of finding element by index to a private method
you can reuse it in many public methods